core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -538,6 +539,14 @@ private[log] class Cleaner(val id: Int,
     for (group <- groupedSegments)
       cleanSegments(log, group, offsetMap, deleteHorizonMs, stats, transactionMetadata)
 
+    if (baseOffset == log.logStartOffset) {
```

Comment:
Why the check? Shouldn't the log cleaner always perform the algorithm in the `if` block?

Reply:
Good catch. The if statement is left from an earlier implementation. I believe cleanable.log will always be the entire log and always have baseOffset be log.logStartOffset.